test/uk/gov/hmrc/minorentityidentificationfrontend/services/SubmissionServiceSpec.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.services
+
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.minorentityidentificationfrontend.controllers.trustControllers.errorControllers.{routes => errorRoutes}
+import uk.gov.hmrc.minorentityidentificationfrontend.helpers.TestConstants._
+import uk.gov.hmrc.minorentityidentificationfrontend.models._
+import uk.gov.hmrc.minorentityidentificationfrontend.services.mocks.{MockStorageService, MockValidateTrustKnownFactsService}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class SubmissionServiceSpec extends AnyWordSpec with Matchers with MockValidateTrustKnownFactsService with MockStorageService {
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  object TestSubmissionService extends SubmissionService(mockValidateTrustKnownFactsService, mockStorageService)
+
+  "submit" should {
+    "return retry url" when {
+      "TrustKnownFacts returns DetailsNotFound, DetailsMismatch or UnMatchableWithRetry" in {
+
+        List(DetailsNotFound, DetailsMismatch).foreach(knownFactsMatchFailure => {
```

Comment:
Test says UnMatchableWithRetry but isn't testing that status

Reply:
ops! The result of the merging .. .going to fix it.